### ISSUE TITLE
REFL: Tidy POLREF configs and add common engineering corrections and distances

### ISSUE THREAD

**Issue by John-Holt-Tessella** [ISISComputingGroup/IBEX#5800]
As an instrument scientist I would like my configs to be clean now we have finished testing them. With a common place with all my engineering corrections in. 

### Acceptance Criteria

- in refl config
    - [x] analyser component is called a1 not m1
    - [ ] no TODOs
    - [x] no commented out sections
    - [x] every motor has a mode dependent engineering corretion
    - [x] mode anr is properly specified
    - [x] engineering corrections and distances are imported from a common file which just holds constants
- [x] Block A1Theta should be A1Phi (check A2Theta)
- [x] Block FINBEAM is added
- [x] CS from horizontal mode is added to vertical mode (currently does not work)
- [x] There are blocks for all vertical config parameters
    - [see here](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/Reflectomtery-IOC-POLREF#specific-axes-by-operation-mode) for blocks on S2 and Bench that need to be renamed
- [ ] Set offset on flipper position (motor is zeroed on reflected beam in SECI)

**Comment by ThomasLohnert** [comment]
Changes on config branch `Ticket5800_tidy_polref_config`. 

Some TODO's remain:
- Set offset on polariser flipper height to zero it on beam (do this in shutdown)
- Set offset on Analyser 1 to zero it on beam (do this in shutdown)
- Adding parked positions & inbeam parameter on Point Detector. (Impeded by #5783). Need to park this manually using axis offsets for now to avoid clashes.

**Comment by DominicOram** [comment]
@John-Holt-Tessella can we pair review this? Just want to check I don't make any glaring errors

**Comment by John-Holt-Tessella** [comment]
Yes, give me a shout when you are ready

**Comment by DominicOram** [comment]
- You've commented out various lambda min/maxes is this 'cos you need functionality to specify beamline constants based on mode? Is so can create a ticket for it? And put a TODO on the comment saying why it's there. Is this also required for MAX_THETA?
- On line 75 of the config where you have `[pnr_name, pa_name]`, it would be nicer to be `[pnr, pa]` and also pull it out into a variable somewhere of eg. polarised_modes. This variable can then be used throughtout
- Can you make a ticket for the s2 tilt correction and move the `s2correction` function on to that ticket and out of the config?
- For the bender you've changed the parked position to 0, isn't this the beam position?
- In the vertical config you have a laser mode, I don't think you should
- Is there a ticket to cover checking the parked position for CS? Can we add it to a ticket for the above TODOs too?
- Can we have a task somewhere to put this on POLREF

**Comment by DominicOram** [comment]
Might be worth a brief note to https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/Reflectometry-Composite-Driving-Layer#select-a-correction-base-on-a-mode about your new helper func too?

**Comment by ThomasLohnert** [comment]
>You've commented out various lambda min/maxes is this 'cos you need functionality to specify beamline constants based on mode? Is so can create a ticket for it? And put a TODO on the comment saying why it's there. Is this also required for MAX_THETA?

Added a comment above the commented out code with the ticket number, added explicit acceptance criteria to #5454

> On line 75 of the config where you have [pnr_name, pa_name], it would be nicer to be [pnr, pa] and also pull it out into a variable somewhere of eg. polarised_modes. This variable can then be used throughtout

`pnr`, `pa` refer to the actual objects for those modes inside the config (rather than their names), but yep have pulled it out into a separate variable

>Can you make a ticket for the s2 tilt correction and move the s2correction function on to that ticket and out of the config?

By this do you mean make the correction part of the reflectometry server? It is a POLREF specific correction, so I think the POLREF config is the right place for it. I have moved it into its own `corrections.py` file for tidiness

> For the bender you've changed the parked position to 0, isn't this the beam position?

Yes 0 should be in beam. This is the parked position I was told by the scientists but they have some axes under SECI where 0 is parked and in beam is non-zero where we have added a user offset in IBEX to make it right under our convention. I have sent the scientists an email to check if this is the case for the BOA, awaiting reply.

> In the vertical config you have a laser mode, I don't think you should

They may want to use the laser to align their samples in vertical mode

> Is there a ticket to cover checking the parked position for CS? Can we add it to a ticket for the above TODOs too?

> Can we have a task somewhere to put this on POLREF

Updated https://github.com/ISISComputingGroup/IBEX/issues/5857 to include all TODO's from here as tasks for POLREF in shutdown except PD park config which is part of the sequential parking ticket

**Comment by DominicOram** [comment]
> By this do you mean make the correction part of the reflectometry server? It is a POLREF specific correction, so I think the POLREF config is the right place for it. I have moved it into its own corrections.py file for tidiness

I meant that we shouldn't have code sitting around that's not used. We should make a ticket to say "let's look at whether we need to use this code" and write the code on the ticket itself.

**Comment by ThomasLohnert** [comment]
Ah I see, yes we will need this code, we are just waiting for the scientists to confirm it is correct before actually sticking the correction into the component. I don't think this is worth the overhead of pulling out the code, creating a ticket, and then merging it in again later

**Comment by DominicOram** [comment]
I think it's worth having the ticket at least, at the moment it's only obvious to you and @John-Holt-Tessella what that code is and what we're waiting on to actually use it
